### ISSUE TITLE
 java.net.ConnectException: Connection refused: connect

### ISSUE THREAD

**Issue by truesrc** [Codeception/CodeceptJS#902]
Hello my friends!
#### What are you trying to achieve?
I want to get the text in the tags `<a>`
#### What do you get instead?
 java.net.ConnectException: Connection refused: connect
> Provide console output if related. Use `--verbose` mode for more details.

Test --
   [1] Starting recording promises
   Emitted | suite.before ([object Object])
   [1] Queued | hook WebDriverIO._beforeSuite()
   [1] Queued | hook REST._beforeSuite()
 Test
   Emitted | test.before
   [1] Queued | hook WebDriverIO._before()
   [1] Queued | hook REST._before()
   Emitted | test.start ([object Object])
   Emitted | step.before (I am on page "http://codecept.io
/helpers/WebDriverIO")
   [1] Queued | hook WebDriverIO._beforeStep()
   [1] Queued | hook REST._beforeStep()
   [1] Queued | amOnPage: "http://codecept.io/helpers/WebD
riverIO"
   Emitted | step.after (I am on page "http://codecept.io/
helpers/WebDriverIO")
   [1] Queued | hook WebDriverIO._afterStep()
   [1] Queued | hook REST._afterStep()
   [1] Queued | step passed
   [1] Queued | return result
   Emitted | step.before (I grab text from "a")
   [1] Queued | hook WebDriverIO._beforeStep()
   [1] Queued | hook REST._beforeStep()
   [1] Queued | grabTextFrom: "a"
   Emitted | step.after (I grab text from "a")
   [1] Queued | hook WebDriverIO._afterStep()
   [1] Queued | hook REST._afterStep()
   [1] Queued | step passed
   [1] Queued | return result
   [1] Queued | create new promises queue for generator
   Emitted | step.start (I am on page "http://codecept.io/
helpers/WebDriverIO")
 * I am on page "http://codecept.io/helpers/WebDriverIO"
   Emitted | step.passed (I am on page "http://codecept.io
/helpers/WebDriverIO")
   Step finished in 6.684 sec
   Emitted | step.start (I grab text from "a")
 * I grab text from "a"
 > [SmartWait (5000ms)] Locating a in 5000
   [1] Error | Error: java.net.ConnectException: Connectio
n refused: connect
   Emitted | step.failed (I grab text from "a")
   Step finished in 1.221 sec
   [1] Error | Error: java.net.ConnectException: Connectio
n refused: connect
   [1] Starting <teardown> session
   Emitted | test.failed ([object Object])
   [1] <teardown> Queued | hook WebDriverIO._failed()
   [1] <teardown> Queued | hook REST._failed()
   [1] <teardown> Queued | () => done(err)
   [1] <teardown> Stopping recording promises
   [1] <teardown> Starting <generator> session

 × FAILED in 10277ms

   [2] Starting recording promises
   Emitted | test.after
   [2] Queued | hook WebDriverIO._after()
   [2] Queued | hook REST._after()
   Emitted | suite.after ([object Object])
   [2] Queued | hook WebDriverIO._afterSuite()
   [2] Queued | hook REST._afterSuite()

-- FAILURES:

  1) Test
       Test:
     java.net.ConnectException: Connection refused: connec
t
  
  Scenario Steps:
  
  - I.grabTextFrom("a") at Test.<anonymous> (D_test.js:9:3
1)
  - I.amOnPage("http://codecept.io/helpers/WebDriverIO") a
t Test.<anonymous> (D_test.js:8:7)
  
  Error: java.net.ConnectException: Connection refused: co
nnect
      at new RuntimeError (C:\Users\Администратор\AppData\
Roaming\npm\node_modules\codeceptjs-webdriverio\node_modul
es\webdriverio\build\lib\utils\ErrorHandler.js:144:12)
      at Request._callback (C:\Users\Администратор\AppData
\Roaming\npm\node_modules\codeceptjs-webdriverio\node_modu
les\webdriverio\build\lib\utils\RequestHandler.js:313:39)
      at Request.self.callback (C:\Users\Администратор\App
Data\Roaming\npm\node_modules\codeceptjs-webdriverio\node_
modules\request\request.js:186:22)
      at Request.<anonymous> (C:\Users\Администратор\AppDa
ta\Roaming\npm\node_modules\codeceptjs-webdriverio\node_mo
dules\request\request.js:1163:10)
      at IncomingMessage.<anonymous> (C:\Users\Администрат
ор\AppData\Roaming\npm\node_modules\codeceptjs-webdriverio
\node_modules\request\request.js:1085:12)

  FAIL  | 0 passed, 1 failed   // 13s
   Emitted | global.result ([object Object])
   [2] Queued | hook WebDriverIO._finishTest()
   [2] Queued | hook REST._finishTest()

```bash
# paste output here
```

> Provide test source code if related

Scenario('Test', function* (I)  {
    I.amOnPage('http://codecept.io/helpers/WebDriverIO');
    let text= yield I.grabTextFrom('a')
})
```js
// paste test
```

### Details

* CodeceptJS version:
* NodeJS Version:
* Operating System:
* Protractor || WebDriverIO || Nightmare version (if related)
* Configuration file:

{
  "tests": "./**/*_test.js",
  "timeout": 100000,
  "output": "./output",
  "helpers": {
    "WebDriverIO": {
      "url": "http://localhost",
      "browser": "chrome",
      "timeouts": {
        "script": 40000,
        "page load": 10000
      },
      "smartWait": 5000
    }
    
    }
  },
  "include": {
    "I": "./steps_file.js"
  },
  "bootstrap": false,
  "mocha": {},
  "name": "tender"
}
Node v8.9.1
CodeceptJS 1.1.0
```js
# paste suite config here
```

**Comment by truesrc** [comment]
Thanks for the answer!
I repeat the test on the problematic scenario > 30 times.
Tests with firefox do not start.
The main thing that I noticed a dependence on the number of tags (possibly any tags) on the page. If the 20 tests pass, if more (30,40,50 ...) fall.

**Comment by reubenmiller** [comment]
So it does sound like the same problem I was seeing. In short, chromedriver doesn't like it when too many requests (through the webdriverio api) are made at the same time.

As for firefox, you probably need to fix something on your standalone selenium server config in order to get it to work with firefox (not a codeceptjs issue).

So I guess we just have to wait for #899 to get merged to test it again.

**Comment by reubenmiller** [comment]
Though if the WebDriverIO Helper api that you are using is also available in the Puppeteer Helper, then you can switch to using Puppeteer (where this problem does not occur)

**Comment by truesrc** [comment]
I test it with CodeceptJS v1.1.2 and "bug" stay.

**Comment by truesrc** [comment]
Puppeteer return the first tag that finds. Not a collection.

**Comment by truesrc** [comment]
I test it with CodeceptJS v1.1.4 and "bug" gone!
Great work boys!
Thank you very much for your attention to the problem!
I hope this has not affected the performance of such a wonderful thing as CodeceptJS.
Meanwhile I found a workaround to just use a more specific query. Like:
`I.grabTextFrom('ul.nav.bs-sidenav') `
and then parse it from string to array, and this way much faster 
`I.grabTextFrom('a') ` in 8691ms
`I.grabTextFrom('ul.nav.bs-sidenav') ` in 4454ms
I opened the issues in the https://youtrack.jetbrains.com/issue/WI-40596 for support CodeceptJS in PhpStorm. 
VOTE ALL!
